ide/server/src/org/netbeans/api/server/CloudProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.api.server;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import org.netbeans.spi.server.ServerInstanceProvider;
+import org.openide.util.lookup.NamedServiceDefinition;
+
+/**
+ * Registration of {@code ServerInstanceProvider} to be displayed under Cloud node in Services.
+ * 
+ * @author Jan Horvath
+ */
+@NamedServiceDefinition(
```

Comment:
Add `apichanges.xml` entry, bump `ide/server` spec version, depend on the new version in `cloud.oracle`

Reply:
plus consider rewrite of other registraitons to use this new annotation.